### PR TITLE
Improve code generation pipeline

### PR DESCRIPTION
**Pull request by EgorkaKulikov**
# Description

This refactoring has two main goals:
- remove all legacy code items from code generator
- improve naming of several classes and methods

Fixes # (issue)

## Type of Change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Requires the retest of the whole code generation pipeline.

### CODE REVIEW

File: utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/generator/CodeGenerationProcessor.kt [hunk 1]
```diff
@@ -61,33 +46,47 @@ import org.jetbrains.kotlin.psi.KtPsiFactory
...
-                        runnable.run()
-                    }
-                }
-                READ_ACTION -> runReadAction { withUtContext(it) { runnable.run() } }
-                WRITE_ACTION -> runWriteAction { withUtContext(it) { runnable.run() } }
-                EDT_LATER -> invokeLater { withUtContext(it) { runnable.run() } }
-            }
-        } ?: error("No context in thread ${Thread.currentThread()}")
-    }
+object CodeGenerationProcessor {
```
Comment (nikitavlaev) [inline utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/generator/CodeGenerationProcessor.kt:88]: Thats more like `CodeGenerationController` for me, but ok

File: utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/models/ExternalLibraryDescriptors.kt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package org.utbot.intellij.plugin.ui
+package org.utbot.intellij.plugin.models
```
Comment (nikitavlaev) [inline utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/models/ExternalLibraryDescriptors.kt:1]: I don't quite understand why this is in the package called `models`.
It seems only related to dialog window, so probably it belongs to `ui`.

File: utbot-framework/src/main/kotlin/org/utbot/framework/plugin/api/TestCaseGenerator.kt [hunk 3]
```diff
@@ -135,128 +117,26 @@ object UtBotTestCaseGenerator : TestCaseGenerator {
...
-            // Update the concrete execution time, if symbolic execution time is small
-            // because of UtSettings.checkSolverTimeoutMillis
-            concreteExecutionBudgetInMillis = userTimeout - symbolicExecutionTimeout
-            require(symbolicExecutionTimeout > 10)
-            require(concreteExecutionBudgetInMillis > 10)
-        }
-    }
-
-    fun generateForSeveralMethods(
+    fun generateTestCases(
```
Comment (nikitavlaev) [inline utbot-framework/src/main/kotlin/org/utbot/framework/plugin/api/TestCaseGenerator.kt:139]: If class is called `'TestCaseGenerator`, maybe this just `generate`?

File: utbot-framework/src/main/kotlin/org/utbot/framework/plugin/api/TestCaseGenerator.kt [hunk 4]
```diff
@@ -24,14 +21,16 @@ import org.utbot.framework.minimization.minimizeTestCase
...
 import org.utbot.framework.plugin.api.util.withUtContext
+import org.utbot.framework.util.jimpleBody
+import org.utbot.framework.util.runSoot
+import org.utbot.framework.util.toModel
 import org.utbot.instrumentation.ConcreteExecutor
 import org.utbot.instrumentation.warmup.Warmup
 import java.io.File
 import java.nio.file.Path
-import java.util.IdentityHashMap
+import java.util.*
```
Comment (nikitavlaev) [inline utbot-framework/src/main/kotlin/org/utbot/framework/plugin/api/TestCaseGenerator.kt:33]: Is wildcard import supposed to be here?

File: utbot-framework/src/test/kotlin/org/utbot/examples/UtModelChecker.kt [hunk 5]
```diff
@@ -136,8 +136,8 @@ internal abstract class AbstractModelBasedTest(
             buildDir = findPathToClassFiles(classLocation)
             previousClassLocation = classLocation
         }
-        UtBotTestCaseGenerator.init(buildDir, classpath = null, dependencyPaths = System.getProperty("java.class.path"))
-        return UtBotTestCaseGenerator.generate(method, mockStrategy)
+        TestCaseGenerator.init(buildDir, classpath = null, dependencyPaths = System.getProperty("java.class.path"))
+        return TestSpecificTestCaseGenerator.generate(method, mockStrategy)
```
Comment (nikitavlaev) [inline utbot-framework/src/test/kotlin/org/utbot/examples/UtModelChecker.kt:140]: Weird behaviour with init of TestCaseGenerator and using TestSpecificTestCaseGenerator requires comment.

File: utbot-framework/src/test/kotlin/org/utbot/examples/UtTestCaseChecker.kt [hunk 6]
```diff
@@ -2493,8 +2493,8 @@ abstract class AbstractTestCaseGeneratorTest(
         mockStrategy: MockStrategyApi,
         additionalDependenciesClassPath: String
     ): UtTestCase {
-        UtBotTestCaseGenerator.init(buildDir, additionalDependenciesClassPath, System.getProperty("java.class.path"))
-        return UtBotTestCaseGenerator.generate(method, mockStrategy)
+        TestCaseGenerator.init(buildDir, additionalDependenciesClassPath, System.getProperty("java.class.path"))
```
Comment (nikitavlaev) [inline utbot-framework/src/test/kotlin/org/utbot/examples/UtTestCaseChecker.kt:2496]: Same comment here (copy-paste here won't hurt anyone)

File: utbot-framework/src/test/kotlin/org/utbot/examples/UtTestCaseChecker.kt [hunk 7]
```diff
@@ -2504,9 +2504,9 @@ abstract class AbstractTestCaseGeneratorTest(
     ): UtTestCase {
         val additionalDependenciesClassPath =
             computeAdditionalDependenciesClasspathAndBuildDir(method, additionalDependencies)
-        UtBotTestCaseGenerator.init(buildDir, additionalDependenciesClassPath, System.getProperty("java.class.path"))
+        TestCaseGenerator.init(buildDir, additionalDependenciesClassPath, System.getProperty("java.class.path"))
         withUtContext(UtContext(method.clazz.java.classLoader)) {
-            return UtBotTestCaseGenerator.generate(method, mockStrategy)
+            return TestSpecificTestCaseGenerator.generate(method, mockStrategy)
```
Comment (nikitavlaev) [inline utbot-framework/src/test/kotlin/org/utbot/examples/UtTestCaseChecker.kt:2509]: Same comment here (copy-paste here won't hurt anyone)

File: utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/generator/CodeGenerationProcessor.kt (hunk 1 above)
Comment (EgorkaKulikov) [reply utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/generator/CodeGenerationProcessor.kt]: Ok, renamed

File: utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/models/ExternalLibraryDescriptors.kt (hunk 2 above)
Comment (EgorkaKulikov) [reply utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/models/ExternalLibraryDescriptors.kt:1]: Actually, it is a model of test framework descriptor in terms of Intellij OpenApi representation. Currently we can their installation only from UI, but actually it is not an unique scenario.

**Review by nikitavlaev — APPROVED** [review]

**Pull request merged**